pkg/products/threescale/reconciler.go

```diff
@@ -704,84 +709,104 @@ func (r *Reconciler) reconcileComponents(ctx context.Context, serverClient k8scl
 		antiAffinityRequired = false
 	}
 
-	key, err := k8sclient.ObjectKeyFromObject(apim)
-	if err != nil {
-		return integreatlyv1alpha1.PhaseFailed, err
-	}
-
+	key := k8sclient.ObjectKeyFromObject(apim)
 	err = serverClient.Get(ctx, key, apim)
 	if err != nil && !k8serr.IsNotFound(err) {
 		return integreatlyv1alpha1.PhaseFailed, err
 	}
 
 	status, err := controllerutil.CreateOrUpdate(ctx, serverClient, apim, func() error {
-
-		apim.Spec.HighAvailability = &threescalev1.HighAvailabilitySpec{Enabled: true}
-		apim.Spec.APIManagerCommonSpec.ResourceRequirementsEnabled = &resourceRequirements
-		apim.Spec.APIManagerCommonSpec.WildcardDomain = r.installation.Spec.RoutingSubdomain
-		apim.Spec.System.FileStorageSpec = fss
-		apim.Spec.PodDisruptionBudget = &threescalev1.PodDisruptionBudgetSpec{Enabled: true}
-		apim.Spec.Monitoring = &threescalev1.MonitoringSpec{Enabled: false}
-
 		replicas := r.Config.GetReplicasConfig(r.installation)
-
-		if *apim.Spec.System.AppSpec.Replicas < replicas["systemApp"] {
-			*apim.Spec.System.AppSpec.Replicas = replicas["systemApp"]
-		}
-		if *apim.Spec.System.SidekiqSpec.Replicas < replicas["systemSidekiq"] {
-			*apim.Spec.System.SidekiqSpec.Replicas = replicas["systemSidekiq"]
-		}
-		if *apim.Spec.Apicast.StagingSpec.Replicas < replicas["apicastStage"] {
-			*apim.Spec.Apicast.StagingSpec.Replicas = replicas["apicastStage"]
-		}
-		if *apim.Spec.Backend.CronSpec.Replicas < replicas["backendCron"] {
-			*apim.Spec.Backend.CronSpec.Replicas = replicas["backendCron"]
-		}
-		if *apim.Spec.Zync.AppSpec.Replicas < replicas["zyncApp"] {
-			*apim.Spec.Zync.AppSpec.Replicas = replicas["zyncApp"]
-		}
-		if *apim.Spec.Zync.QueSpec.Replicas < replicas["zyncQue"] {
-			*apim.Spec.Zync.QueSpec.Replicas = replicas["zyncQue"]
+		systemAppReplicas := replicas["systemApp"]
+		systemSidekiqReplicas := replicas["systemSidekiq"]
+		apicastStageReplicas := replicas["apicastStage"]
+		backendCronReplicas := replicas["backendCron"]
+		zyncReplicas := replicas["zyncApp"]
+		zyncQueReplicas := replicas["zyncQue"]
+		apim.Spec = threescalev1.APIManagerSpec{
```

Comment:
I would not do this in this manner. You are rewriting this block very time. Are all the field in the CR spec managed by rhoam?

Reply:
Sorry I don't understand what is the question. 
What's your recommendation? 
RHOAM sets the spec of the apimanager.